manifests/renew.pp

```diff
@@ -86,13 +90,52 @@
   ]).filter | $arg | { $arg =~ NotUndef and $arg != [] }
   $command = join($_command, ' ')
 
-  cron { 'letsencrypt-renew':
-    ensure   => $cron_ensure,
-    command  => $command,
-    user     => 'root',
-    hour     => $cron_hour,
-    minute   => $cron_minute,
-    monthday => $cron_monthday,
+  case $timer_type {
+    'cron': {
+      cron { 'letsencrypt-renew':
+        ensure   => $cron_ensure,
+        command  => $command,
+        user     => 'root',
+        hour     => $cron_hour,
+        minute   => $cron_minute,
+        monthday => $cron_monthday,
+      }
+    }
+    'systemd': {
+      if $facts['systemd'] {
```

Comment:
The fact use should ne like this:
```suggestion
       if $facts['service_provider'] == 'systemd' {
```
In the `if` block, we do configuration stuffs. In the `else` block, i understand we are in a situation without systemd available and we are trying to use it. So `else` block should return error with `fail()`.

Reply:
my suggestion here is to use the selector pattern:
https://puppet.com/docs/puppet/latest/lang_conditional.html#selectors

Reply:
I believe I have fixed this to match what both of you have requested.